### PR TITLE
Adding subdialog for Scatterplot

### PR DESCRIPTION
**Pull request by maxwellfundi**
@dannyparsons have a look at this. I have linked the subdialog by seting the R syntax for the subdialog and eting up layer. 
I am not sure if more is need. 
Is there?

### CODE REVIEW

File: instat/dlgScatterPlot.vb [hunk 1]
```diff
@@ -85,6 +85,8 @@ Public Class dlgScatterPlot
         ucrFactorOptionalReceiver.Selector = ucrSelectorForScatter
         ucrFactorOptionalReceiver.SetIncludedDataTypes({"factor"})
 
+        sdgLayerOptions.SetRSyntax(ucrBase.clsRsyntax)
```
Comment (dannyparsons) [inline instat/dlgScatterPlot.vb]: This isn't needed anymore

File: instat/dlgScatterPlot.vb [hunk 2]
```diff
@@ -120,6 +122,7 @@ Public Class dlgScatterPlot
     End Sub
 
     Private Sub cmdScatterPlotOptions_Click(sender As Object, e As EventArgs) Handles cmdScatterPlotOptions.Click
+        sdgLayerOptions.SetupLayer(clsTempGgPlot:=clsRggplotFunction, clsTempGeomFunc:=clsRgeom_scatterplotFunction, clsTempAesFunc:=clsRaesFunction, bFixAes:=True, bFixGeom:=True, strDataframe:=ucrSelectorForScatter.ucrAvailableDataFrames.cboAvailableDataFrames.Text, bUseGlobalAes:=True)
         sdgLayerOptions.ShowDialog()
```
Comment (dannyparsons) [inline instat/dlgScatterPlot.vb]: You need to also do what I have done for boxplot after this, which is syncing the receivers on the dialog.

Comment (maxwellfundi) [inline instat/dlgScatterPlot.vb]: for all receivers?

Comment (dannyparsons) [inline instat/dlgScatterPlot.vb]: Yes as I did for boxplot. If you have changed something on the subdialog it should reflect in the main dialog receivers.

File: instat/dlgScatterPlot.vb (hunk 1 above)
Comment (maxwellfundi) [inline instat/dlgScatterPlot.vb]: Okay

**Pull request merged**